packages/btcstaking/src/staking.rs

```diff
@@ -0,0 +1,830 @@
+use crate::errors::Error;
+use crate::Result;
+
+use crate::types::is_rate_valid;
+use babylon_schnorr_adaptor_signature::{verify_digest, AdaptorSignature};
+use bitcoin::hashes::Hash;
+use bitcoin::opcodes::all::OP_RETURN;
+use bitcoin::sighash::{Prevouts, SighashCache};
+use bitcoin::{Script, Transaction, TxOut};
+use k256::schnorr::Signature as SchnorrSignature;
+use k256::schnorr::VerifyingKey;
+
+/// Maximum transaction weight allowed in Babylon system.
+/// This matches the MaxStandardTxWeight constant from Babylon Genesis.
+const MAX_STANDARD_TX_WEIGHT: usize = 400000;
+
+/// Maximum transaction version allowed in Babylon system.
+/// This matches the maxTxVersion constant from Babylon Genesis.
+const MAX_TX_VERSION: i32 = 2;
+
+/// Dust threshold defines the maximum value of an output to be considered a dust output.
+const DUST_THRESHOLD: u64 = 546;
+
+/// Checks if a script is an OP_RETURN output
+fn is_op_return_output(script: &bitcoin::ScriptBuf) -> bool {
+    let script_bytes = script.as_bytes();
+    !script_bytes.is_empty() && script_bytes[0] == OP_RETURN.to_u8()
+}
+
+/// Checks pre-signed transaction sanity
```

Comment:
It's a private function and this comment basically repeats the function name, adding little value.
```suggestion
```